### PR TITLE
Create RTL demo for marketing

### PR DESCRIPTION
**Pull request by yuriy-fix**
Fixes https://github.com/vaadin/components-team-tasks/issues/522

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/bookstore-example/81)

### CODE REVIEW

**Comment by Haprog** [comment]
Minor thing but this doesn't look right to me:
image
I guess the snapshot version string should be wrapped in something like:

`<span dir="ltr">3.1-SNAPSHOT</span>`

But I'm not 100% sure if this is expected or not that the version string parts split by hyphen are shown in different order. At least it can still be copy-pasted and it will show correctly in LTR context it seems.

**Comment by Haprog** [comment]
Looks pretty good to me otherwise, but some minor UX improvement could be made on the language switcher so it's easier to find in case you don't understand the currently selected language.

For example when I first tried this I selected the Persian/Farsi language and logged in it took me a while to realize that you can switch the language back to English from the top bar.

Since there are only 2 languages to choose from, most clear imo would be to show them both as text (buttons or links) next to each other like "[English]() | [فارسی]()". Then it's very easy to spot the word "English" from the top even if you first get to the other language version. You could even hide the active language option, but it might make more sense to show both (for more context if you wouldn't otherwise know that the weird text string is the name of another language). Maybe having the active one with a different style (either some active style, or disabled style).

Another alternative would be to keep the dropdown/select but add some kind of language selector icon next to it. Something like a globe icon or something similar to what Wikipedia uses: 
image


See: https://usersnap.com/blog/design-language-switch/

File: src/main/java/org/vaadin/example/bookstore/ui/about/AboutView.java [hunk 1]
```diff
@@ -8,19 +9,20 @@
...
 @PageTitle("About")
 public class AboutView extends HorizontalLayout {
 
-    public static final String VIEW_NAME = "About";
+    private transient ResourceBundle resourceBundle = ResourceBundle.getBundle("MockDataWords", UI.getCurrent().getLocale());
 
     public AboutView() {
         add(VaadinIcon.INFO_CIRCLE.create());
-        add(new Span(" This application is using Vaadin version "
+        add(new Span(" " + resourceBundle.getString("application_sign") + " "
```
Comment (shadikhani) [inline src/main/java/org/vaadin/example/bookstore/ui/about/AboutView.java:22]: In Farsi it is should be written like this:
```
add(new Span(“این برنامه از وادین نسخه ”
        + Version.getFullVersion() + ” استفاده می کند.“));
```

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java [hunk 2]
```diff
@@ -99,18 +103,18 @@ public ProductForm(InventoryViewLogic sampleCrudLogic) {
...
+        productName = new TextField(resourceBundle.getString("product_name"));
         productName.setWidth("100%");
         productName.setRequired(true);
         productName.setValueChangeMode(ValueChangeMode.EAGER);
         content.add(productName);
 
-        price = new TextField("Price");
+        price = new TextField(resourceBundle.getString("price"));
         price.setSuffixComponent(new Span("€"));
         price.addThemeVariants(TextFieldVariant.LUMO_ALIGN_RIGHT);
```
Comment (shadikhani) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java:114]: All number field alignments should be left.
In the enum file TextFieldVariant, should have LUMO_ALIGN_LEFT.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java [hunk 3]
```diff
@@ -99,18 +103,18 @@ public ProductForm(InventoryViewLogic sampleCrudLogic) {
...
         price.setValueChangeMode(ValueChangeMode.EAGER);
 
-        stockCount = new TextField("In stock");
+        stockCount = new TextField(resourceBundle.getString("in_stock"));
         stockCount.addThemeVariants(TextFieldVariant.LUMO_ALIGN_RIGHT);
```
Comment (shadikhani) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java:118]: stockCount text-align should be left.

File: src/main/resources/ValidationMessages_fa_IR.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+size.category_characters_msg=\u0646\u0627\u0645 \u0637\u0628\u0642\u0647 \u0628\u0646\u062f\u06cc \u0628\u0627\u06cc\u062f \u062d\u062f\u0627\u0642\u0644 \u062f\u0648 \u062d\u0631\u0641 \u062f\u0627\u0634\u062a\u0647 \u0628\u0627\u0634\u062f
+size.product_characters_msg=\u0646\u0627\u0645 \u0645\u062d\u0635\u0648\u0644 \u0628\u0627\u06cc\u062f \u062d\u062f\u0627\u0642\u0644 \u062f\u0648 \u062d\u0631\u0641 \u062f\u0627\u0634\u062a\u0647 \u0628\u0627\u0634\u062f
+min.negative_amount_msg=\u0646\u0645\u06cc \u062a\u0648\u0627\u0646 \u0645\u0628\u0644\u063a \u0645\u0646\u0641\u06cc \u062f\u0631 \u0633\u0647\u0627\u0645 \u062f\u0627\u0634\u062a
```
Comment (shadikhani) [inline src/main/resources/ValidationMessages_fa_IR.properties:3]: Validation messages are still in English

File: src/main/java/org/vaadin/example/bookstore/ui/about/AboutView.java (hunk 1 above)
Comment (yuriy-fix) [reply src/main/java/org/vaadin/example/bookstore/ui/about/AboutView.java]: Updated.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java (hunk 2 above)
Comment (yuriy-fix) [reply src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java]: Updated.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java (hunk 3 above)
Comment (yuriy-fix) [reply src/main/java/org/vaadin/example/bookstore/ui/inventory/ProductForm.java]: Updated.

File: src/main/resources/ValidationMessages_fa_IR.properties (hunk 4 above)
Comment (yuriy-fix) [reply src/main/resources/ValidationMessages_fa_IR.properties]: Updated.

File: pom.xml [hunk 5]
```diff
@@ -17,7 +17,7 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <failOnMissingWebXml>false</failOnMissingWebXml>
 
-        <vaadin.version>15.0.5</vaadin.version>
+        <vaadin.version>16.0-SNAPSHOT</vaadin.version>
```
Comment (DiegoCardoso) [inline pom.xml:20]: I am not sure if you should use `16.0.0.beta1` here or leave it as snapshot.
```suggestion
        <vaadin.version>16.0.0.beta1</vaadin.version>
```

File: pom.xml [hunk 6]
```diff
@@ -158,7 +158,7 @@
             <plugin>
                 <groupId>com.vaadin</groupId>
                 <artifactId>vaadin-maven-plugin</artifactId>
-                <version>${vaadin.version}</version>
+                <version>16.0.0.alpha2</version>
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply pom.xml]: Bumped now.

File: pom.xml (hunk 5 above)
Comment (yuriy-fix) [reply pom.xml]: Bumped now.

**Pull request merged**